src/jenkins.py

```diff
@@ -288,47 +278,94 @@ def _unlock_wizard(container: ops.Container) -> None:
 
     Args:
         container: The Jenkins workload container.
+
+    Raises:
+        JenkinsBootstrapError: if the wizard can not be unlocked.
     """
-    version = get_version()
-    container.push(
-        LAST_EXEC_VERSION_PATH,
-        version,
-        encoding="utf-8",
-        make_dirs=True,
-        user=USER,
-        group=GROUP,
-    )
-    container.push(
-        WIZARD_VERSION_PATH,
-        version,
-        encoding="utf-8",
-        make_dirs=True,
-        user=USER,
-        group=GROUP,
-    )
+    try:
+        version = get_version()
+        container.push(
+            LAST_EXEC_VERSION_PATH,
+            version,
+            encoding="utf-8",
+            make_dirs=True,
+            user=USER,
+            group=GROUP,
+        )
+        container.push(
+            WIZARD_VERSION_PATH,
+            version,
+            encoding="utf-8",
+            make_dirs=True,
+            user=USER,
+            group=GROUP,
+        )
+    except (ops.pebble.PathError, JenkinsError) as exc:
+        raise JenkinsBootstrapError("Failed to unlock wizard.") from exc
 
 
-def _install_configs(container: ops.Container) -> None:
+def _install_config(container: ops.Container, filename: str, destination_path: Path) -> None:
     """Install jenkins-config.xml.
 
+    Args:
+        container: The Jenkins workload container.
+        filename: the source file to copy contents from.
+        destination_path: the target path.
+
+    Raises:
+        JenkinsBootstrapError: if the config can not be installed.
+
+    """
+    try:
+        with open(filename, encoding="utf-8") as jenkins_config_file:
```

Comment:
I think this can be made into `pathlib.Path` type and use `read_text()` for file contents to help remove the with clause :)